### ISSUE TITLE
Unable to find EVS commandline utilities

### ISSUE THREAD

**Issue by cybermac72** [snorre-k/idrive-docker#6]
Thank you again for your responsiveness to previous issues identified.  I wanted to share that a couple of users on a different forum have reported the following error upon pulling your docker container following the fix for the TZ variable:

![image png a02e34591d87dc4a656104ba82ff92dc](https://github.com/snorre-k/idrive-docker/assets/82063746/2b598fb1-e718-413f-b8b8-5a90c10fff70)

I have been unable to replicate this issue as yet so I'm not sure how or why it happens but one occurrence can be explained away while two occurrences from separate individuals is a bit odd.

**Comment by snorre-k** [comment]
After some time of testing (therefore I write this late), I was able to reproduce the error.

The problem arises, when the volume `idrive` mounted to `/opt/IDriveForLinux/idriveIt` is partially empty or corrupt.
The volume usually gets filled with the content of this directory in the container at the first start, or when it is empty.
After account configuration, the backup schedules and settings as well as the account credentials are stored here.

Resolution for this problem is to delete the volume and create a new one. After that, a new logon and creation of the schedules is necessary.

I am unsure, how the 'corruption' of the volume can happen. Maybe you can get more information about the issue in your forum?

**Comment by cybermac72** [comment]
> After some time of testing (therefore I write this late), I was able to reproduce the error.
> 
> The problem arises, when the volume `idrive` mounted to `/opt/IDriveForLinux/idriveIt` is partially empty or corrupt. The volume usually gets filled with the content of this directory in the container at the first start, or when it is empty. After account configuration, the backup schedules and settings as well as the account credentials are stored here.
> 
> Resolution for this problem is to delete the volume and create a new one. After that, a new logon and creation of the schedules is necessary.
> 
> I am unsure, how the 'corruption' of the volume can happen. Maybe you can get more information about the issue in your forum?

Hmm, I am left wondering about the possibility then of when launching the container for the first time -- if the volume mapping for the host side DID NOT exist (in other words, they were letting the volume be created automatically by docker)??

Could that be it?

Docker would create them -- but as you said they would be "empty".  Is it sufficient to simply create the folder/directory before launching the container to avoid the error OR does an empty idrivecrontab.json need to be there too?

**Comment by snorre-k** [comment]
There is a volume definition in the docker build file, which automatically creates a local volume, when a volume definition is not specified. This volume is named with a large UUID:
```
docker volume ls
DRIVER    VOLUME NAME
local     37eece927c9108f655931e5676ad9de06a214c58f6d71994fa34394e5d370e1f
local     idrive-docker_idrive
```
Also this volume is filled automatically as it is completely empty. If you put one file in the volume (e.g. via another container), it will not be filled and covers / overlaps the content of the docker folder at this mountpoint.

The problem is, that configuration data and programs are in the same folder `/opt/IDriveForLinux/idriveIt`. This is a design problem for container volumes, which can only be solved by the iDrive developers. But I think this peace of software was never thought to be run in a container :)

So to sum it up:

- The problem only exists when program files from this folder get deleted or do not exist.
- The volume is filled automatically with the files in the containers folder, when **NO** files exist in the volume
- The volume gets created automatically, when no volume configuration is specified.

**Comment by cybermac72** [comment]
I've reached out to query the two individuals on their attempts, but what you suggest seems more than plausible.  I'll let you know if anything said differs from the expected.

**Comment by cybermac72** [comment]
I have not heard anything back from the two individuals.  I know they were able to resolve their issues by running the update again.  I suspect their issues were due to what we hypothesized.  I was not able to reproduce it myself, so might as well go ahead and resolve this issue.  Thank you again for your attention, assistance, and receptiveness with resolving these issues!

**Comment by snorre-k** [comment]
I will close the issue now. THX for reporting.
